src/fb.d.ts

```diff
@@ -0,0 +1,39 @@
+interface FBOptions {
+    accessToken: string;
+    appId: string;
+    appSecret: string;
+    version?: string;
+    proxy?: string;
+    timeout?: number;
+    scope?: string;
+    redirectUri?: string;
+    Promise?: Promise;
+}
+
+export class Facebook {
+    constructor(options: FBOptions);
+
+    api(...args): Promise<any>;
+
+    extend(options: FBOptions): Facebook;
+
+    getAccessToken(): string;
+
+    getLoginUrl(): string;
+
+    napi(...args): Promise<any>;
+
+    options(keyOrOptions: FBOptions): FBOptions;
+
+    parseSignedRequest(signedRequest: string, ...args): Object;
+
+    setAccessToken(accessToken: string): void;
+
+    withAccessToken(accessToken: string): any;
+}
+
+export const version: string;
+
+export function FacebookApiException(res: Response): void;
```

Comment:
Where is `Response` defined? `void` also is wrong, this is a subclass of `Error`.

Reply:
`Response` is from TypeScript lib.dom.d.ts.